pulp_rpm/app/modulemd.py

```diff
@@ -100,10 +100,15 @@ def create_modulemd(modulemd, snippet):
     new_module[PULP_MODULE_ATTR.STATIC_CONTEXT] = modulemd["data"].get("static_context")
     new_module[PULP_MODULE_ATTR.CONTEXT] = modulemd["data"].get("context")
     new_module[PULP_MODULE_ATTR.ARCH] = modulemd["data"].get("arch")
-    new_module[PULP_MODULE_ATTR.ARTIFACTS] = modulemd["data"].get("artifacts", [])
+    new_module[PULP_MODULE_ATTR.ARTIFACTS] = modulemd["data"].get("artifacts", {}).get("rpms", [])
     new_module[PULP_MODULE_ATTR.DESCRIPTION] = modulemd["data"].get("description")
     new_module[PULP_MODULE_ATTR.PROFILES] = modulemd["data"].get("profiles", {})
-    new_module[PULP_MODULE_ATTR.DEPENDENCIES] = modulemd["data"].get("dependencies", [])
+    for dep in modulemd["data"].get("dependencies", []):
+        if "requires" in dep:
+            new_module[PULP_MODULE_ATTR.DEPENDENCIES] = dep["requires"]
+            break
+    else:
+        new_module[PULP_MODULE_ATTR.DEPENDENCIES] = []
```

Comment:
This sets .DEPENDENCIES to an empty-list **unless** we have `modulemd["data"]["dependencies"]["requires"][stuff]` - is that correct? Should the *if* not have an else-clause `new_module[PULP_MODULE_ATTR.DEPENDENCIES] = dep` ?

Reply:
The problem is that https://github.com/fedora-modularity/libmodulemd/blob/main/yaml_specs/modulemd_stream_v2.yaml#L210 seems to get parsed as a list.  So instead of `dependencies = {"requires": ..., "buildrequires": ...}` you get `dependencies = [{"requires": ...}, {"buildrequires": ...}]`

So it's treating it as a list of which one item will be the dict we're looking for, and then extract the deps from that one.

Reply:
Oh, and also prior to this we were ending up with `dependencies = [{}]` somehow.

Reply:
> seems to get parsed as a list.

I'd have expected it to be `dependencies = [{"requires":..., "buildrequires":....}, {"requires":..., "buildrequires":...}, ...]`

I think the (or at least my) confusion is that the linked YAML is used to **build** the same module for different platforms using a single yaml - hence the list-of buildrequires/requires . By the time *we* see the result, there is only one - the one for the "platform" we're syncing. (Yes?) And we don't care about "buildrequires" at that point. 

So we only ever get a list with one dictionary-entry, and we only care about the existence of the "requires" key therein.

Side-note from looking closely at the YAML: Do we need to care about "extras" and "moreextras"?

Reply:
> Side-note from looking closely at the YAML: Do we need to care about "extras" and "moreextras"?

nm those are inside the dict we end up with.

Reply:
We've never cared about build data in the past.  That it exists at all is just because the format mixes together "what clients need" with "what MBS needs"

It's not even present in our old snippets much less dependencies